applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/explicit_central_differences_scheme.hpp

```diff
@@ -564,12 +568,13 @@ class ExplicitCentralDifferencesScheme
                 r_current_acceleration = zero_array;
             }
 
-            bool fix_displacements[3] = {false, false, false};
+            std::array<bool, 3> fix_displacements = {false, false, false};
 
-            fix_displacements[0] = (it_node->pGetDof(DISPLACEMENT_X))->IsFixed();
-            fix_displacements[1] = (it_node->pGetDof(DISPLACEMENT_Y))->IsFixed();
+            const IndexType disppos = it_node->GetDofPosition(DISPLACEMENT_X);
```

Comment:
have this out of the loop, this is the expensive function...

actually you may even want to store disppos and rotpos as class members...

Reply:
I thought was something like the EquationId, so is common in the variable list?, OK, updating

Reply:
Done